pkg/daemon/update.go

```diff
@@ -556,6 +572,15 @@ func (dn *Daemon) updateKernelArguments(oldConfig, newConfig *mcfgv1.MachineConf
 	return exec.Command("rpm-ostree", args...).Run()
 }
 
+// updateToRealTimeKernel updates to RT Kernel if it is set to true.
+// In case realtime is set to false, switch to traditional Kernel if already
+// on RT kernel, otherwise do nothing
+func (dn *Daemon) updateToRealTimeKernel(oldConfig, newConfig *mcfgv1.MachineConfig) error {
```

Comment:
Can we change the function signature to something clearer\*? It gets confusing when you have `updateToRealTimeKernel(oldConfig, newConfig)` then `updateToRealTimeKernel(newConfig, oldConfig)` in the revert where the new is the old and the old is the new 😟 .

Best would be to take only the minimal information from the configs necessary for the function.

\* https://dave.cheney.net/2019/09/24/be-wary-of-functions-which-take-several-parameters-of-the-same-type